static/js/hotspots.js

```diff
@@ -2,41 +2,183 @@ var hotspots = (function () {
-function mark_hotspot_as_read(hotspot) {
+function post_hotspot_as_read(hotspot) {
     channel.post({
         url: '/json/users/me/hotspots',
-        data: {hotspot: JSON.stringify(hotspot)},
+        data: { hotspot: JSON.stringify(hotspot.name) },
+        error: function (err) {
+            blueslip.error(err.responseText);
+        },
     });
 }
 
-$(function () {
-    $("#hotspot_click_to_reply").on('click', function (e) {
-        mark_hotspot_as_read("click_to_reply");
+function attach_hotspot_events(hotspot) {
+    var hotspot_id = '#hotspot_' + hotspot.name;
+    var popover_selector = hotspot_id + ' .hotspot-popover-wrapper';
+
+    // open
+    $(hotspot_id + ' .hotspot-icon').on('click', function (e) {
         e.preventDefault();
         e.stopPropagation();
+
+        // hide icon
+        $(this).animate({ opacity: 0 }, {
+            duration: 300,
+            done: function () {
+                $(this).css({ display: 'none' });
+            }.bind(this),
+        });
+
+        // position popover
+        $(hotspot_id + ' .hotspot-popover').css({
+            display: 'block',
+            top: hotspot.location.popover.top,
+            left: hotspot.location.popover.left,
+        });
+
+        // position arrow
+        $(hotspot_id + ' .hotspot-popover').addClass(
+            'arrow-' + hotspot.location.popover.arrow_placement
+        );
+
+        // show popover
+        $(popover_selector).css({
+            display: 'block',
+        }).animate(
+            { opacity: 1 },
+            { duration: 300 }
+        );
     });
-    $("#hotspot_new_topic_button").on('click', function (e) {
-        mark_hotspot_as_read("new_topic_button");
+
+    // confirm
+    $(hotspot_id + ' .hotspot-confirm').on('click', function (e) {
         e.preventDefault();
         e.stopPropagation();
+
+        post_hotspot_as_read(hotspot);
+
+        $(hotspot_id).animate({ opacity: 0 }, {
+            duration: 200,
+            done: function () {
+                $(hotspot_id).remove();
+            }.bind(this),
+        });
     });
-    $("#hotspot_stream_settings").on('click', function (e) {
-        mark_hotspot_as_read("stream_settings");
-        e.preventDefault();
+
+    // close
+    $('body').on('click', function () {
+        var close_hotspot =
+            !$(event.target).closest('.hotspot').length &&
+            $(popover_selector).css('display') === 'block';
+
+        if (close_hotspot) {
+            $(popover_selector).animate({
+                opacity: 0,
+            }, {
+                duration: 300,
+                done: function () {
+                    $(popover_selector).css({ display: 'none' });
+
+                    $(hotspot_id + ' .hotspot-icon').css({
+                        display: 'inline',
+                    }).animate(
+                        { opacity: 1 },
+                        { duration: 300 }
+                    );
+                },
+            });
+        }
+    });
```

Comment:
Moved the functionality for closing the hotspot popovers into `hotspots.hide_hotspot_popovers`, which is invoked in `popovers.hide_all`.

I've kept the `$(hotspot_id + ' .hotspot-icon')` click handlers as they are. Registering those clicks as `$('body').on('click', '.hotspot .hotspot-icon')` is inconsistent because parent elements that the hotspots are appended to can intercept propagation.